Extract OrdersController#populate into RESTful action

gsmendoza opened the pull request:
## Background

This PR is part of a broader task to make actions from OrdersController RESTful.

## Previous implementations

See https://github.com/solidusio/solidus/pull/4102.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- ~[ ] Bug fix (non-breaking change which fixes an issue)~
- ~[ ] New feature (non-breaking change which adds functionality)~
- ~[ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)~

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- ~[ ] My change requires a change to the documentation.~
- ~[ ] I have updated the documentation accordingly.~

gsmendoza commented on templates/spec/requests/spree/orders_spec.rb:
```diff
@@ -6,81 +6,6 @@
   let!(:store) { create(:store) }
   let(:variant) { create(:variant) }
 
-  context "#populate" do
```
I deleted the `populate` specs because 1) they display the deprecation warning and 2) they're already tested by the `OrderContentsController#create` specs.

kennyadsl commented on templates/app/controllers/spree/orders_controller.rb:
```diff
@@ -77,6 +77,8 @@ def populate
         end
       end
     end
+    deprecate populate: 'please use OrdersContentController#create instead',
```
I think we can just remove this. Being a template, deprecating code is not needed.

gsmendoza replied:
Hehe.. You're right. I'll remove the deprecation notice. TODO.

gsmendoza commented:
@kennyadsl I updated the PR

* Removed `populate_redirect`.
* Removed the deprecation commit.
* Squashed the commits replacing `OrdersController#populate` with `OrderContentsController#create`.

waiting-for-dev commented on templates/app/controllers/spree/order_contents_controller.rb:
```diff
@@ -0,0 +1,47 @@
+module Spree
```
Missing `frozen-string-literals` comment 🙂

gsmendoza replied:
Oops! Missed that one. Fixed!

waiting-for-dev commented on templates/config/routes.rb:
```diff
@@ -46,11 +48,9 @@
   get '/checkout/:state', to: 'checkout#edit', as: :checkout_state
   get '/checkout', to: 'checkout#edit', as: :checkout
 
-  get '/orders/populate', to: 'orders#populate_redirect'
```
No replacement for that?

kennyadsl replied:
Yes, Solidus is not using this route. It was added because people may refresh the browser page when adding to the cart in some weird way (maybe this is not more possible with recent browsers?) causing a 404 to happen. I think we can avoid this.

waiting-for-dev reviewed (commented):
Thanks, @gsmendoza, some comments 🙂

gsmendoza commented:
@waiting-for-dev Updated the PR:

* Added the missing `frozen_string_literal` comment.

kennyadsl reviewed (approved):
👍 I suspect some rebasing pain merging this before #246. Up to how how to handle the merge order!

waiting-for-dev reviewed (approved):
Thanks @gsmendoza !
